### ISSUE TITLE
EarlyStopping auto mode is incorrect for AUC

### ISSUE THREAD

**Issue by eli-osherovich** [keras-team/keras#15138]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): no
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Debian
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 2.6
- Python version: 3.9
- Bazel version (if compiling from source): 4.0
- GPU model and memory: NA
- Exact command to reproduce: NA

**Describe the problem**.
It seems that the `EarlyStopping` callback does not recognize the correct mode for the AUC metric.
Instead of tracking `max` it tracks `min`

**Describe the current behavior**.
The best iteration chosen is the one with lowest AUC.

**Describe the expected behavior**.
The best iteration is the one with highest AUC.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

**Standalone code to reproduce the issue**.

Create any model with AUC metric, set `EarlyStopping` to track `val_auc` and run `model.fit()`. The training will stop after the minimal AUC (instead of maximal).

**Comment by ymodak** [comment]
Can you please provide a reproducible example to illustrate the behavior? Thanks!

**Comment by DachuanZhao** [comment]
I pull a new request for it . Thanks ~

**Comment by ymodak** [comment]
Closing since the associated PR is merged. Thanks!
